### ISSUE TITLE
Index error using v2.2b5 after refinement step

### ISSUE THREAD

**Issue by kretsge** [DeepLabCut/DeepLabCut#815]
OS: Windows 10
DeepLabCut Version: 2.2b5
Anaconda env used: DLC-GPU

Hi! I've just started to transition to version 2.2 for multi-animal tracking. I've been working in the Project Manager (though I previously used older versions of DLC in a jupyter notebook).

I set up my first multi-animal project and it worked through the annotation and training of the network. I refined several videos because there were some instances when it switched which mouse was which and I think that also went smoothly. 

However, after I did this refinement in the GUI, I attempted to merge the dataset and re-train the network. When I tried to train the network, I got an Index Error. I'm not sure how to interpret this error or how to figure out what went wrong. I'm unsure if I made a mistake in the refinement GUI, in merging the datasets, or in attempting to train the network. 

Do you have advice about how to figure out exactly what's going wrong using the Project Manager GUI? Or in terminal? The screenshot below is from the terminal window when this happened. I had tried to restart the GUI and then train the network with iteration 1. Thanks in advance for your help!

<img width="987" alt="Screen Shot 2020-07-06 at 4 47 59 PM" src="https://user-images.githubusercontent.com/22033541/87988732-cff0b380-caae-11ea-900e-8cfbe0f3c58a.png">

**Comment by jeylau** [comment]
Hi @kretsge! Could you post screenshots of your config.yaml? It looks like there is something wrong with TrainFraction.

**Comment by kretsge** [comment]
Hi! Here are the screenshots of my config.yaml file. Let me know what you think.

<img width="789" alt="Screen Shot 2020-07-21 at 11 47 39 AM (1)" src="https://user-images.githubusercontent.com/22033541/88348201-cb303780-cd1a-11ea-994b-b7c4ceb8dc75.png">
<img width="574" alt="Screen Shot 2020-07-21 at 11 47 55 AM 2" src="https://user-images.githubusercontent.com/22033541/88348205-cd929180-cd1a-11ea-90e4-7312ec3de7f6.png">
<img width="244" alt="Screen Shot 2020-07-21 at 11 48 08 AM 3" src="https://user-images.githubusercontent.com/22033541/88348209-cff4eb80-cd1a-11ea-8b0e-665fe1229b5a.png">
<img width="271" alt="Screen Shot 2020-07-21 at 11 48 18 AM 4" src="https://user-images.githubusercontent.com/22033541/88348212-d2574580-cd1a-11ea-87ae-a1d258582621.png">
<img width="986" alt="Screen Shot 2020-07-21 at 11 48 35 AM5" src="https://user-images.githubusercontent.com/22033541/88348217-d4210900-cd1a-11ea-9eee-d111fc5ab041.png">

**Comment by jeylau** [comment]
@kretsge, what did you mean above by training with iteration 1? Was `trainsetindex`=0?

**Comment by kretsge** [comment]
Maybe this is a misunderstanding on my end. When I used older versions of DLC, after I went through the refinement step and merged datasets, when I re-ran “train network” it would automatically update to iteration 1. Should that not be what happens here after refining the data and merging the datasets?

**Comment by MMathisLab** [comment]
hi @kretsge thanks for the added details, indeed it should be as you say, i.e. after merge data it bumps up to 1 and it should retrain. I will check a few things and get back to you!

**Comment by kretsge** [comment]
Ok, thank you so much! Let me know if you need any other information from me.

**Comment by kretsge** [comment]
Hi- I just wanted to check back in about this issue in case it got lost in the shuffle.  Thanks for your help!

**Comment by MMathisLab** [comment]
Hi @kretsge apologies; I couldn't reproduce the issues, but want to try one more thing! Will be in touch this week 👍

**Comment by MMathisLab** [comment]
hi @kretsge I double checked, and had no issues :/

Let me be clear on my workflow: created a new project, labeled 4 frames, trained for 200 iterations, ran video analysis and tracking GUI to get an .h5 file. Then, extracted 4 outliers, refined them in the GUI, clicked merge, which takes you back to create-training dataset tab, created set, and then went to train again... 

```
Merged data sets and updated refinement iteration to 1.
Now you can create a new training set for the expanded annotated images (use create_training_dataset).
/Users/mwmathis/Desktop/testing-mwm-2020-08-16/labeled-data/videocompressed11short_cropped  already exists!
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:17<00:00,  4.32s/it]
Utilizing the following graph: [[0, 1], [1, 2], [1, 3], [2, 3], [0, 3], [0, 2]]
Creating training data for  1 0.95
This can take some time...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:00<00:00, 43.60it/s]
The training dataset is successfully created. Use the function 'train_network' to start training. Happy training!
Selecting multi-animal trainer
Config:
{'all_joints': [[0], [1], [2], [3]],
 'all_joints_names': ['bodypart1', 'bodypart2', 'bodypart3', 'bodypart4'],
 'batch_size': 8,
 'crop_pad': 0,
 'cropratio': 0.4,
 'dataset': 'training-datasets/iteration-1/UnaugmentedDataSet_testingAug16/testing_mwm95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'deterministic': False,
 'display_iters': 500,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/Users/mwmathis/Documents/DeepLabCut/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'max_input_size': 1500,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets/iteration-1/UnaugmentedDataSet_testingAug16/Documentation_data-testing_95shuffle1.pickle',
 'min_input_size': 64,
 'mirror': False,
 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]],
 'net_type': 'resnet_50',
 'num_joints': 4,
 'num_limbs': 6,
 'optimizer': 'adam',
 'pafwidth': 20,
 'pairwise_huber_loss': False,
 'pairwise_loss_weight': 0.1,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1], [1, 2], [1, 3], [2, 3], [0, 3], [0, 2]],
 'partaffinityfield_predict': True,
 'pos_dist_thresh': 17,
 'project_path': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16',
 'regularize': False,
 'rotation': 25,
 'rotratio': 0.4,
 'save_iters': 10000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16/dlc-models/iteration-1/testingAug16-trainset95shuffle1/train/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Activating limb prediction...
Starting with multi-animal imaug + adam pose-dataset loader.
Batch Size is 8
Getting specs multi-animal-imgaug 6 4
Initializing ResNet
Loading ImageNet-pretrained resnet_50
2020-08-16 16:51:57.421438: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Max_iters overwritten as 45
Display_iters overwritten as 1
Save_iters overwritten as 14
Training parameters:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16/dlc-models/iteration-1/testingAug16-trainset95shuffle1/train/snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'adam', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 8, 'dataset_type': 'multi-animal-imgaug', 'deterministic': False, 'mirror': False, 'pairwise_huber_loss': False, 'weigh_only_present_joints': False, 'partaffinityfield_predict': True, 'pairwise_predict': True, 'all_joints': [[0], [1], [2], [3]], 'all_joints_names': ['bodypart1', 'bodypart2', 'bodypart3', 'bodypart4'], 'cropratio': 0.4, 'dataset': 'training-datasets/iteration-1/UnaugmentedDataSet_testingAug16/testing_mwm95shuffle1.pickle', 'display_iters': 500, 'init_weights': '/Users/mwmathis/Documents/DeepLabCut/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt', 'max_input_size': 1500, 'metadataset': 'training-datasets/iteration-1/UnaugmentedDataSet_testingAug16/Documentation_data-testing_95shuffle1.pickle', 'min_input_size': 64, 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]], 'net_type': 'resnet_50', 'num_joints': 4, 'num_limbs': 6, 'pafwidth': 20, 'pairwise_loss_weight': 0.1, 'partaffinityfield_graph': [[0, 1], [1, 2], [1, 3], [2, 3], [0, 3], [0, 2]], 'pos_dist_thresh': 17, 'project_path': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16', 'rotation': 25, 'rotratio': 0.4, 'save_iters': 10000, 'scale_jitter_lo': 0.5, 'scale_jitter_up': 1.25}
Starting multi-animal training....
iteration: 1 loss: 1.6941 scmap loss: 1.5092 locref loss: 0.0313 limb loss: 0.1536 lr: 0.0001
```
I would suggest bumping up to the latest, which would require you to just run the following in the env:

`pip install git+https://github.com/DeepLabCut/DeepLabCut.git`

**Comment by MMathisLab** [comment]
Then I tried the other workflow, or supplemental workflow: i.e., I grabbed edited frames from the Tracklet GUI. I.e., if you "drag" any points in the GUI you can save those edited frames to throw back into the dataset. So here, I click "optional merge refined frames" and then went to the refine labels tab, clicked MERGE," which bumps it up to 2, then I create training dataset, and train again:

```
Frames from video videocompressed11short  already extracted (more will be added)!
A training dataset file is already found for this video. The refined machine labels are merged to this data!
Merged data sets and updated refinement iteration to 2.
Now you can create a new training set for the expanded annotated images (use create_training_dataset).
Utilizing the following graph: [[0, 1], [1, 2], [1, 3], [2, 3], [0, 3], [0, 2]]
Creating training data for  1 0.95
This can take some time...
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 30/30 [00:00<00:00, 43.05it/s]
The training dataset is successfully created. Use the function 'train_network' to start training. Happy training!
Selecting multi-animal trainer
Config:
{'all_joints': [[0], [1], [2], [3]],
 'all_joints_names': ['bodypart1', 'bodypart2', 'bodypart3', 'bodypart4'],
 'batch_size': 8,
 'crop_pad': 0,
 'cropratio': 0.4,
 'dataset': 'training-datasets/iteration-2/UnaugmentedDataSet_testingAug16/testing_mwm95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'deterministic': False,
 'display_iters': 500,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/Users/mwmathis/Documents/DeepLabCut/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'max_input_size': 1500,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets/iteration-2/UnaugmentedDataSet_testingAug16/Documentation_data-testing_95shuffle1.pickle',
 'min_input_size': 64,
 'mirror': False,
 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]],
 'net_type': 'resnet_50',
 'num_joints': 4,
 'num_limbs': 6,
 'optimizer': 'adam',
 'pafwidth': 20,
 'pairwise_huber_loss': False,
 'pairwise_loss_weight': 0.1,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1], [1, 2], [1, 3], [2, 3], [0, 3], [0, 2]],
 'partaffinityfield_predict': True,
 'pos_dist_thresh': 17,
 'project_path': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16',
 'regularize': False,
 'rotation': 25,
 'rotratio': 0.4,
 'save_iters': 10000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16/dlc-models/iteration-2/testingAug16-trainset95shuffle1/train/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Activating limb prediction...
Starting with multi-animal imaug + adam pose-dataset loader.
Batch Size is 8
Getting specs multi-animal-imgaug 6 4
Initializing ResNet
Loading ImageNet-pretrained resnet_50
Max_iters overwritten as 45
Display_iters overwritten as 1
Save_iters overwritten as 14
Training parameters:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16/dlc-models/iteration-2/testingAug16-trainset95shuffle1/train/snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'adam', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 8, 'dataset_type': 'multi-animal-imgaug', 'deterministic': False, 'mirror': False, 'pairwise_huber_loss': False, 'weigh_only_present_joints': False, 'partaffinityfield_predict': True, 'pairwise_predict': True, 'all_joints': [[0], [1], [2], [3]], 'all_joints_names': ['bodypart1', 'bodypart2', 'bodypart3', 'bodypart4'], 'cropratio': 0.4, 'dataset': 'training-datasets/iteration-2/UnaugmentedDataSet_testingAug16/testing_mwm95shuffle1.pickle', 'display_iters': 500, 'init_weights': '/Users/mwmathis/Documents/DeepLabCut/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt', 'max_input_size': 1500, 'metadataset': 'training-datasets/iteration-2/UnaugmentedDataSet_testingAug16/Documentation_data-testing_95shuffle1.pickle', 'min_input_size': 64, 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]], 'net_type': 'resnet_50', 'num_joints': 4, 'num_limbs': 6, 'pafwidth': 20, 'pairwise_loss_weight': 0.1, 'partaffinityfield_graph': [[0, 1], [1, 2], [1, 3], [2, 3], [0, 3], [0, 2]], 'pos_dist_thresh': 17, 'project_path': '/Users/mwmathis/Desktop/testing-mwm-2020-08-16', 'rotation': 25, 'rotratio': 0.4, 'save_iters': 10000, 'scale_jitter_lo': 0.5, 'scale_jitter_up': 1.25}
Starting multi-animal training....
iteration: 1 loss: 1.6079 scmap loss: 1.4150 locref loss: 0.0303 limb loss: 0.1627 lr: 0.0001
iteration: 2 loss: 0.5056 scmap loss: 0.4667 locref loss: 0.0074 limb loss: 0.0316 lr: 0.0001
iteration: 3 loss: 0.3601 scmap loss: 0.2870 locref loss: 0.0130 limb loss: 0.0601 lr: 0.0001
iteration: 4 loss: 0.2367 scmap loss: 0.1047 locref loss: 0.0243 limb loss: 0.1077 lr: 0.0001
iteration: 5 loss: 0.1774 scmap loss: 0.0688 locref loss: 0.0286 limb loss: 0.0800 lr: 0.0001
```

**Comment by kretsge** [comment]
Thank you for your response! I decided to start a new project from scratch and see if I ran into the same issue. I'm just about to refine the labels so I wanted to double check with you that I'm doing this correctly.

I did the "refine tracklets" step and clicked "optional: merge refined data" for a few different videos (merging after each video). Next I extracted outlier frames and went to the "refine labels" tab. When I launched the GUI I saw that I had these videos with hundreds of frames to label, which seems like a lot (see attached image). 

So a few questions:
1) Should I be merging the data/creating new datasets/training the network after refining the tracklets? Or can I refine tracklets, merge the data, refine the labels, merge that data, and then create a new training dataset with all of that together? 2) When I used the older (not multi-animal) version of DLC I would only do about 20 frames per video when refining. Is there a way to not refine the hundreds of frames per video that I've extracted? Is there a way to get rid of these frames that I've already extracted without messing up the data?

Again, thank you so much for your help. I hope this new project I've started will work and there was just some error I made the first time around!

<img width="998" alt="Screen Shot 2020-08-17 at 12 04 17 PM" src="https://user-images.githubusercontent.com/22033541/90417503-ca709400-e081-11ea-8308-40fe5df7968e.png">
